OptionSelector/CQDetailTableViewController.m

```diff
@@ -0,0 +1,119 @@
+//
+//  CQDetailTableViewController.m
+//  Unit1WeekFourHomework
+//
+//  Created by Jackie Meggesto on 8/12/15.
+//  Copyright (c) 2015 Jackie Meggesto. All rights reserved.
+//
+
+#import "CQDetailTableViewController.h"
+
+@interface CQDetailTableViewController ()
+@property (nonatomic) NSIndexPath *checkedIndexPath;
+
+
+@end
+
+@implementation CQDetailTableViewController
+
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    
+    
+    
+    
+    // Uncomment the following line to preserve selection between presentations.
+    // self.clearsSelectionOnViewWillAppear = NO;
+    
+    // Uncomment the following line to display an Edit button in the navigation bar for this view controller.
+    // self.navigationItem.rightBarButtonItem = self.editButtonItem;
+}
+
+-(void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
+    self.checkedIndexPath = indexPath;
+    self.category.cqIndexPath = indexPath;
+    self.category.selection = self.category.options[indexPath.row];
+    [self.tableView reloadData];
+}
+
+- (void)didReceiveMemoryWarning {
+    [super didReceiveMemoryWarning];
+    // Dispose of any resources that can be recreated.
+}
+
+#pragma mark - Table view data source
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
+#warning Potentially incomplete method implementation.
+    // Return the number of sections.
+    return 1;
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+#warning Incomplete method implementation.
+    // Return the number of rows in the section.
+    return self.category.options.count;
+}
+
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"DetailCell" forIndexPath:indexPath];
+    
+    cell.textLabel.text = self.category.options[indexPath.row];
+    if ([self.checkedIndexPath isEqual:indexPath] || [self.category.cqIndexPath isEqual:indexPath]) {
```

Comment:
In order to obviate the `cqIndexPath` property on `CQCategory`, you can just compare the option at the current index path to the category's `selection` property like so:
```objc
NSString *option = self.category.options[indexPath.row];
if ([self.category.selection isEqualToString:option]) {
    // set accessory type to checkmark
} else {
    // set accessory type to none
}
```